src/core/cache.rs

```diff
@@ -0,0 +1,91 @@
+use crate::core::error::Error;
+
+use clru::CLruCache;
+use cosmwasm_vm::{Checksum, Size};
+use wasmer::{Instance, Module, Store};
+
+const ESTIMATED_MODULE_SIZE: Size = Size::mebi(10);
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Stats {
+    pub hits: u32,
+    pub misses: u32,
+}
+
+impl Stats {
+    pub fn new() -> Self {
+        Self { hits: 0, misses: 0 }
+    }
+}
+
+/// An in-memory module cache
+pub struct InMemoryCache {
+    modules: CLruCache<Checksum, Module>,
+}
+
+impl InMemoryCache {
+    /// Creates a new cache with the given size (in bytes)
+    pub fn new(size: Size) -> Self {
+        let max_entries = size.0 / ESTIMATED_MODULE_SIZE.0;
+        InMemoryCache { modules: CLruCache::new(max_entries) }
+    }
+
+    pub fn store(&mut self, checksum: &Checksum, module: &Module) -> Result<(), Error> {
```

Comment:
module should be `Module` (store should be the ownership of this clone inside function seems to be the wrong pattern)